python/python/lance/dataset.py

```diff
@@ -94,8 +94,13 @@
 class LanceDataset(pa.dataset.Dataset):
     """A dataset in Lance format where the data is stored at the given uri."""
 
-    def __init__(
-        self,
+    def __init__(self, _ds: _Dataset):
+        self._uri = _ds.uri
+        self._ds = _ds
```

Comment:
This is going to break so much existing code...

Could we instead just make a classmethod and keep the existing init method:

```python
@classmethod
def _from_inner(cls, ds: _Dataset):
    self = cls.__new__()
    self._uri = _ds.uri
    self._ds = _ds
    return self
```